### ISSUE TITLE
ci: fix failing ghcr image publish jobs

### ISSUE THREAD

**Issue by petermetz** [hyperledger/cactus#1881]
**Describe the bug**

The problem is described in the email I'm pasting below:

---
On Thu, Dec 30, 2021 at 09:18 Somogyvari, Peter <[peter.somogyvari@accenture.com](mailto:peter.somogyvari@accenture.com)> wrote:
Hi Community Architects,

I'm at the finish line with migrating Cactus away from DockerHub to GHCR and hit on something that I think might be about org level permissions somehow (just a guess). Below is my attempt to clarify the symptoms, could you please advise if you think it's possible that some of our images are configure differently (in GH org settings that I cannot see with my permissions)
1.	The same workflow yaml files are working fine on the hyperledger-cicd org (where I am an admin)
2.	On the "production" org (e.g. [github.com/hyperledger](https://urldefense.proofpoint.com/v2/url?u=http-3A__github.com_hyperledger&d=DwMFaQ&c=eIGjsITfXP_y-DLLX0uEHXJvU8nOHrUK8IrwNKOtkVU&r=7LpfyaMR44exzQCWvxiAAfpAoMMDWWRNgSJjU7B7o4c&m=GNHUH_O2KisLywj9sWAXXkJJ27gcPyYxIYbwr2jKqRbFaWarX9-Lla7CKJ5Sf3DE&s=uyE8m9MsBrHSs5hxDwUs9F-CfmognOoPQzYhA1cEMg8&e=)) the same yaml files that are publishing containers are only partially working
a.	So yes, some of them work fine, but others fail to publish the built image with permission problems
3.	The error for the ones that fail the git push step is this: denied: permission_denied: write_package
4.	An example of a failing one is this image: [ghcr.io/hyperledger/cactus-besu-all-in-one](https://urldefense.proofpoint.com/v2/url?u=http-3A__ghcr.io_hyperledger_cactus-2Dbesu-2Dall-2Din-2Done&d=DwMFaQ&c=eIGjsITfXP_y-DLLX0uEHXJvU8nOHrUK8IrwNKOtkVU&r=7LpfyaMR44exzQCWvxiAAfpAoMMDWWRNgSJjU7B7o4c&m=GNHUH_O2KisLywj9sWAXXkJJ27gcPyYxIYbwr2jKqRbFaWarX9-Lla7CKJ5Sf3DE&s=wfRgrt62VYuc-N5re6qWcT_HIUV1V3BIqC9iZdSRG_E&e=)
5.	An example of one that works just fine is [ghcr.io/hyperledger/cactus-connector-fabric](https://urldefense.proofpoint.com/v2/url?u=http-3A__ghcr.io_hyperledger_cactus-2Dconnector-2Dfabric&d=DwMFaQ&c=eIGjsITfXP_y-DLLX0uEHXJvU8nOHrUK8IrwNKOtkVU&r=7LpfyaMR44exzQCWvxiAAfpAoMMDWWRNgSJjU7B7o4c&m=GNHUH_O2KisLywj9sWAXXkJJ27gcPyYxIYbwr2jKqRbFaWarX9-Lla7CKJ5Sf3DE&s=wrajgfKN-GiUQqGr0YQsspGgoLurXZ4RXM1j_d5zR6c&e=)
6.	I'm attaching the CI logs for both above-mentioned image publishing tasks named as success.log and failure.log for reference (but really the only thing I can see in it is as meaningful information is text in red above)
7.	The publishing is done the same way for both images, the yaml files are copy pasted with the only difference being the dockerfiles' path and the image tag.
a.	[https://github.com/hyperledger/cactus/blob/6a3107f87e630d35bb088d223edb36eb20fb64e9/.github/workflows/besu-all-in-one-publish.yaml](https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_hyperledger_cactus_blob_6a3107f87e630d35bb088d223edb36eb20fb64e9_.github_workflows_besu-2Dall-2Din-2Done-2Dpublish.yaml&d=DwMFaQ&c=eIGjsITfXP_y-DLLX0uEHXJvU8nOHrUK8IrwNKOtkVU&r=7LpfyaMR44exzQCWvxiAAfpAoMMDWWRNgSJjU7B7o4c&m=GNHUH_O2KisLywj9sWAXXkJJ27gcPyYxIYbwr2jKqRbFaWarX9-Lla7CKJ5Sf3DE&s=KupldbCbvx9_rpfHJz1-eujztEPAD_LCHHhJk-envX8&e=)
b.	[https://github.com/hyperledger/cactus/blob/6a3107f87e630d35bb088d223edb36eb20fb64e9/.github/workflows/connector-fabric-publish.yaml](https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_hyperledger_cactus_blob_6a3107f87e630d35bb088d223edb36eb20fb64e9_.github_workflows_connector-2Dfabric-2Dpublish.yaml&d=DwMFaQ&c=eIGjsITfXP_y-DLLX0uEHXJvU8nOHrUK8IrwNKOtkVU&r=7LpfyaMR44exzQCWvxiAAfpAoMMDWWRNgSJjU7B7o4c&m=GNHUH_O2KisLywj9sWAXXkJJ27gcPyYxIYbwr2jKqRbFaWarX9-Lla7CKJ5Sf3DE&s=va_M5f9EgvBiSNhlRrOvrhVlvqEyZ_XnarDi2eWcX9k&e=)
---

**To Reproduce**

Steps to reproduce the behavior on a successfully deployed Hyperledger Cactus cluster.

**Expected behavior**

Image publishing should work fine for every commit that gets merged to `main`

**Logs/Stack traces**

See above in the email.

**Screenshots**

![image](https://user-images.githubusercontent.com/1583231/155622564-e1d1b295-85ee-449e-8afd-bde68324e0e2.png)

![image](https://user-images.githubusercontent.com/1583231/155622719-442f01c7-9648-453f-ba6a-aa0e2a8f85e1.png)

**Cloud provider or hardware configuration:**

GitHub Actions - Workflow Runner

**Operating system name, version, build:**

Ubuntu-20.04

**Hyperledger Cactus release version or commit (git rev-parse --short HEAD):**

1.0.0-rc.3

**Hyperledger Cactus Plugins/Connectors Used**

N/A

**Additional context**

This is a follow-up to https://github.com/hyperledger/cactus/issues/995
which was the original issue for implementing GHCR publishing to begin with.

**Comment by petermetz** [comment]
Dumping my latest email on the topic here for reference:

---

Hi Ry,

I just realized you probably meant that I would add the workflow dispatch. 
/facepalm

I just did the following:
1.	Invited you to my fork as a collaborator (just in case it helps)
2.	Started a branch on my fork that has the dispatch requested and some extra patterns in case you want to experiment with your own branches as well: https://github.com/petermetz/cactus/tree/petermetz/issue1881
3.	Also pushed the same branch to the upstream (hyperledger/cactus) which is here: https://github.com/hyperledger/cactus/tree/petermetz/issue1881
4.	Noticed that on my fork the publishing job passes fine (image tag is ghcr.io/petermetz/cactus-…) while the same branch on the upstream repo fails with the permission error I mentioned earlier (ghcr.io/hyperledger/cactus-…).

Does this help with the debugging? I’m only hoping that I understood your request correctly. 😊

Kind regards,

---
